### ISSUE TITLE
Facebook share doesn't work

### ISSUE THREAD

**Issue by Cheomm88** [consul/consul#2643]
# Steps to reproduce
1- Enter in any bugdet investment.
2- In the share with social networks click in facebook
3- The facebook popup returns an error and user can't share the element with Facebook

# Expected behaviour
- The popup should show the Facebook UI and allow the user to share the item.

# Current Behaviour
- The popup's URL have the "quote" empty if it gets the descreption of the item it will work. This was fixed in the website from Madrid and added to the consul 0.14  but it stills happens.

**Comment by decabeza** [comment]
Hi @Cheomm88,

This issue was fixed [in this commit](https://github.com/consul/consul/pull/2525/commits/d4b88f1fae583f8f7769dbee939b2048bee01356).

Can you confirm what environment you're testing in? In a local environment, sharing with Facebook does not work. 😔 But it does work in production.

**Locally still appearing this `&quote=%20`**
![screen shot 2018-05-23 at 10 53 58](https://user-images.githubusercontent.com/631897/40414314-2be8d500-5e78-11e8-8a70-6ce9a2c18695.png)

**Work correctly on production environment**
![screen shot 2018-05-23 at 10 57 26](https://user-images.githubusercontent.com/631897/40414309-2a2e0adc-5e78-11e8-8fda-760ac4f39fe5.png)

**Comment by Cheomm88** [comment]
We are with 0.14 in our "PRE release" enviroment, our "Producion enviroment" is still in the 0.13 so I can't tell you more.

But in slack more people had this issue.

**Comment by decabeza** [comment]
@Cheomm88 can you confirm if you "inspect" facebook button on the browser have all this `data` attributes? 

```shell
<div class="social-share-button" 
data-title="" 
data-img=""
data-url="" 
data-desc="" 
data-via="">
```

Thanks! ☺️

**Comment by Cheomm88** [comment]
@decabeza

**Comment by decabeza** [comment]
I think the problem is on the empty `data-desc=""`. 🤔 

On `app/views/budgets/investments/_investment_show.html.erb` L164, do you have `description: investment.title`?

**Comment by Cheomm88** [comment]
I'm using a custom view, but passing to the partial the ` description: investment.title` but still doesn't work.

**Comment by decabeza** [comment]
@Cheomm88 do you have a public repo to take a look at the code? Or a live environment to test it? 👀

**Comment by andosasesores** [comment]
@Cheomm88 @decabeza  Hi, you probably need to update the file "app/views/shared/_social_share.html.erb" and add the attribute "desc" in the helper "social_share_button_tag". Here's an example of how it should look:

```ruby
<% if local_assigns[:share_title].present? %>
  <div id="social-share" class="sidebar-divider"></div>
  <p class="sidebar-title"><%= share_title %></p>
<% end %>
<div class="social-share-full">
  <%= social_share_button_tag("#{title} #{setting['twitter_hashtag']}",
                              :url => local_assigns[:url],
                              :image => local_assigns[:image_url].present? ? local_assigns[:image_url] : '',
                              desc: local_assigns[:description] ) %>
  <a href="whatsapp://send?text=<%= CGI.escape(title) %>&nbsp;<%= url %>"
     class="show-for-small-only" data-action="share/whatsapp/share">
    <span class="icon-whatsapp whatsapp"></span>
    <span class="show-for-sr"><%= t("social.whatsapp") %></span>
  </a>
</div>
```

**Comment by javierm** [comment]
Closing since the proposed solution seemed to work and the fix was on master since pull request #2525.
